java/src/main/java/org/whispersystems/signalservice/api/SignalServiceMessageSender.java

```diff
@@ -170,18 +170,30 @@ public void sendMessage(SignalServiceAddress recipient, SignalServiceDataMessage
   public void sendMessage(List<SignalServiceAddress> recipients, SignalServiceDataMessage message)
       throws IOException, EncapsulatedExceptions
   {
-    byte[]              content   = createMessageContent(message);
-    long                timestamp = message.getTimestamp();
-    SendMessageResponse response  = sendMessage(recipients, timestamp, content, true);
+    byte[]                  content                 = createMessageContent(message);
+    long                    timestamp               = message.getTimestamp();
+    EncapsulatedExceptions  encapsulatedExceptions  = null;
+
+    try {
+      sendMessage(recipients, timestamp, content, true);
+    } catch (EncapsulatedExceptions e) {
+      encapsulatedExceptions = e;
+    }
 
     try {
-      if (response != null && response.getNeedsSync()) {
+      if (!store.getSubDeviceSessions(localAddress.getNumber()).isEmpty()) {
```

Comment:
The "normal" behaviour in case of NetworkFailureExceptions is saving them to disk to send them later, in case of UntrustedIdentotyExceptions or UnregisteredUserExceptions it is [ignoring them](https://github.com/WhisperSystems/Signal-Android/blob/9148b7da5f4604dae1106d3932bfa7d3bfa63ad4/src/org/thoughtcrime/securesms/notifications/MessageNotifier.java#L317).

Groups with unregistered users will always happen until we get a proper group administration. And even then, it will most likely not guaranteed that your view of the group matches the server's reality and this will still be a problem.